docs/services/qix-engine/access-control.md

```diff
@@ -141,78 +159,83 @@ In addition to this `user` will contain all other attributes defined in the
+#### Type-specific resource attributes
 
-Depending on what type of resource that is being accessed, `resource` may carry additional attributes. If the resource
-type is `"App"`, there are no additional attributes. If the resource type is `"App.Object"`, `resource` has the
-following additional attributes:
+Depending on the type of resource that is being accessed, `resource` may carry additional attributes.
+If the resource type is `"App"`, there are no additional attributes.
+If the resource type is `"App.Object"`, then `resource` has the following additional attributes:
 
 | Attribute | Description | Example condition |
 | --------- | ----------- | ----------------- |
 | `resource._objecttype` | The object type. | `resource._objecttype = "field" or resource._objecttype = "my-generic-object"` |
 | `resource.app` | Reference to the app resource that the object is part of. | `resource.app.owner = "john-doe"` |
 
-##### The `app` Attribute
+##### The `app` attribute
 
-As stated above, if `resource._resourcetype = "App.Object"`, `resource.app` contains the reference to the `"App"`
-resource that the object exists in, and all attributes and [built-in functions](#built-in_function) that apply
-to `"App"` objects are available on `resource.app`.
+When the common resource attribute is `resource._resourcetype = "App.Object"`,
+then the type-specific resource attribute `resource.app` contains a reference to the `app` resource.
+This means that all attributes and built-in functions are available on `resource.app`.
 
-#### Built-in Functions
+#### `HasPrivilege` built-in function
 
-##### `HasPrivilege(<action>)`
+If the user making the request has already been granted access for the provided action,
+then the built-in `HasPrivilege(<action>)` function returns `true`, otherwise it returns `false`.
 
-Boolean function that returns `true` if the user making the request has already been granted access for the provided
-action. Otherwise returns `false`.
-
-**Syntax**
+The syntax of the build-in function is the following:
 
 ```c
 resource.HasPrivilege(ACTION)
 ```
 
-The required parameter `ACTION` shall have a value of any of the supported actions described in [Actions](#Actions).
+The `ACTION` parameter is required. Its value can be any of the supported actions described in [Actions](#actions).
 
-**Examples**
+Consider the following rules:
 
 ```c
 user.country = "uk" and resource._resourcetype = "App.Object" and resource._actions = "create"
 resource._resourcetype = "App.Object" and resource.HasPrivilege("create") and resource._actions = {"read", "update"}
 ```
 
-Here, the second rule uses `HasPrivilege` to check if the `create` action has already been granted to the user, which
-could be result from the preceeding rule. If that is the case, it also grants access to the `read` and `update` actions.
+The first rule grants the `create` action to a user.
+
+The second rules has a built-in function that will evaluate to `true` if
```

Comment:
Maybe refer directly to `HasPrivilege` instead of "... has a built-in function ...".